### DISCUSSION TITLE
Is there a chance to support managed identity in the iot-sdk APIs?

### DISCUSSION THREAD

**Discussion by ranjith-t** [Azure/azure-iot-sdk-java#1655] [Ideas]
In the aure-iot-sdk-java apart from TokenCredential and AzureSasCredential, I could not see API support with managed identity for secure connection. Is there a plan to include them in the future?

**Comment by timtay-microsoft** [comment]
Currently, we support managed identity credentials for authentication in the IoT hub service client as seen in [this sample code](https://github.com/Azure/azure-iot-sdk-java/blob/a5b5a1fa1020ac8e833b827d7dc4e9296edcbc52/service/iot-service-samples/role-based-authorization-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/RoleBasedAuthenticationSample.java#L66). The TokenCredential class allows you to pass in a number of implementations from the Azure Identity package. For more examples of this, see [here](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/samples/java/com/azure/identity/credential/JavaDocCodeSnippets.java).

We don't have any plans to add similar managed identity support on the IoT hub device client because the service itself does not support that kind of authentication yet.

Hope that helps!
-Tim

**Reply by ranjith-t** [reply]
@timtay-microsoft , thank you for the references. we are able to do service app registration and token based auth for services(client app). As far as I understand, these tokens has expiry, can you please guide what is the best practice for renewal, if this is on-demand or time based?

**Reply by timtay-microsoft** [reply]
In general, the renewal is taken care of for you by the Azure Identity package itself. Whether a given type of authentication (interactive login vs hardcoded tenantId + clientId + client secret) has a strict time limit would be a better question for the folks maintaining [that library](https://github.com/Azure/azure-sdk-for-java/discussions).
